### ISSUE TITLE
torch.norm does not work when p="nuc"

### ISSUE THREAD

**Issue by LeeJZh** [pytorch/pytorch#18275]
## 🐛 Bug

torch.norm does not work as expected when p="nuc" , x is a 4 dimensional tensor, dim=(2,3)
## To Reproduce

`
import torch

w = torch.randn(2, 1024, 3, 3)

torch.norm(w, p="fro", dim=(2,3)) ## this works fine

torch.norm(w, p="nuc", dim=(2,3)) ## this produce an error message
`

RuntimeError                              Traceback (most recent call last)
<ipython-input-4-88fe1a07ad43> in <module>()
----> 1 torch.norm(w, p="nuc", dim=(2,3))

~/anaconda3/lib/python3.6/site-packages/torch/functional.py in norm(input, p, dim, keepdim, out)
    726     elif p == "nuc":
    727         if out is None:
--> 728             torch._C._VariableFunctions.nuclear_norm(input, keepdim=keepdim)
    729         return torch._C._VariableFunctions.nuclear_norm(input, keepdim=keepdim, out=out)
    730     else:

RuntimeError: Expected a tensor with 2 dimensions, but got a 4 dimensions tensor instead.

## Expected behavior
return the track norm.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): 1.0.0
 - OS (e.g., Linux): Ubuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.6.8
 - CUDA/cuDNN version: 10.0
 - GPU models and configuration: Two Titan XP
 - Any other relevant information:

## Additional context
I also run the above code with Pytorch 1.0.1.post2, the error is the same.

**Comment by soumith** [comment]
good find, this is a bug. We need to compute it on the last two dimensions.

**Comment by vishwakftw** [comment]
I have two thoughts:

1. this is not a bug, but rather a limitation of the linear algebra backend. I believe ‘nuc’ uses the SVD internally, which is only implemented for 2D tensors.
2. this is a bug considering dims is not passed to the nuclear norm implementation as indicated by the traceback.

**Comment by LeeJZh** [comment]
It seems that dim is not passed to csrc when p="nuc".
[torch.norm source](https://pytorch.org/docs/stable/_modules/torch/functional.html#norm)
`
elif p == "nuc":

        if out is None:

            torch._C._VariableFunctions.nuclear_norm(input, keepdim=keepdim)

        return torch._C._VariableFunctions.nuclear_norm(input, keepdim=keepdim, out=out)
`

**Comment by skrah** [comment]
Done in #21022.

**Comment by canbakiskan** [comment]
When torch.norm is used with `dim=(2,3)` and `p='nuc'` the gradient cannot be computed because U and V  in SVD are not computed. 

Full error:
> RuntimeError: svd_backward: Setting compute_uv to false in torch.svd doesn't compute singular matrices, and hence we cannot compute backward. Please use torch.svd(compute_uv=True)


Setup:
Pytorch version: 1.3.0.dev20190914 (nightly)
OS (e.g., Linux): Ubuntu 18.04
How you installed PyTorch (conda, pip, source): conda
Python version: 3.6.8
CUDA/cuDNN version: 10.1.243 / 7.6

Code:
```
import torch

matrix1 = torch.ones(32,32)
matrix1.requires_grad = True

nuc_norm = torch.norm(matrix1, p='nuc')

nuc_norm.backward() # works fine


tensor1 = torch.ones(10,3,32,32)
tensor1.requires_grad = True

nuc_norm = torch.norm(tensor1, p='nuc', dim=(2,3))

loss = nuc_norm.sum()

loss.backward() # U, V not computed error
```

In aten/src/ATen/native/LinearAlgebra.cpp functions` nuclear_norm()` and `nuclear_norm_out()` uses `_batch_svd()` with `compute_uv=False`. Maybe this could be made optional somehow.

**Comment by canbakiskan** [comment]
I suggest changing the argument `/*compute_uv=*/false` in functions `nuclear_norm()` and `nuclear_norm_out()` in aten/src/ATen/native/LinearAlgebra.cpp to `/*compute_uv=*/self.requires_grad()`. I tested it (not using the official test files, I don't know how to do that) on both cpu and cuda10.0 with current source code build and it does work.

**Comment by vishwakftw** [comment]
@canbakiskan The fix is available in #26303, thank you for reporting.
